bip-rpob-tx-template.mediawiki

```diff
@@ -0,0 +1,284 @@
+
+<code>HASH256</code> is defined by ''double sha256'':
+
+<pre>
+rpob_revocation_puzzle:    32-bytes, public, SHA256(SHA256(
+                                     <32-byte-rpob_revocation_solution>
+                                     ))
+
+rpob_revocation_solution:  32-bytes, secret, publicly reveal this value to revoke the proof-of-burn.
+</pre>
+
+*It is RECOMMENDED that implementations use a value for the <code>rpob_revocation_solution</code> that is derived from the secret key used to generate the  <code>rpob_secp256k1_public_key</code> .
+
+===Output 3, Optional: Change===
+<pre>
+value:              (any)
+public_key_script:  1 <32-byte-taproot_witness_program>
+</pre>
+
+*This output MAY have any value.
+*The public key MUST be a standard P2TW output.
+
+===Signature: Included in Output 1===
+<pre>
+rpob_schnorr_signature:       64-bytes, bip-340, signature (
+                                        <32-byte-rpob_signature_message_hash>,
+                                        <32-byte-secp256k1_private_key>
+                                        )
+
+rpob_signature_message_hash:  32-bytes, bip-340, sha256_taged_hash (
+                                        <77-byte-rpob_signature_message>,
+                                        <16-byte-rpob_hash_tag>
+                                        )
+
+rpob_hash_tag:                16-bytes, utf-8-hash_tag ("rpob/BTC/MAINNET")
+                                    or, utf-8-hash_tag ("rpob/BTC/TESTNET")
+
+rpob_signature_message:       77-bytes, < 1-byte-version_flags>
+                                        <32-byte-input_txid_hash>
+                                        < 4-byte-input_index>
+                                        < 8-byte-burn_output_value>
+                                        <32-byte-rpob_revocation_puzzle>
+</pre>
+
+*The <code>rpob_schnorr_signature</code> uses the [https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki BIP-340] algorithm.
+
+==Deployment==
+
+While the ''IsStandard'' rules are quite restricting, it is quite possible to submit transactions directly to miners, or co-operate with miners who would enjoy to have some addition fee-revenue. So the initial process of testing on the main-network should be possible.
+
+If this standard gains significant attention, we are happy to write a supplementary BIP to define a new service bit to allow nodes to signal that this new type of standard transaction is accepted.
+
+==Rationale==
+===Why require RPoB transactions to use a new transaction version?===
+
+This allows for future indexing and enforcement strategies at the consensus layer.
+
+===Why use a hash puzzle for the revocation?===
+
+The hash-puzzle allows for easy indexing of the revocation, in the format:
+
+<code><32-byte-revocation-puzzle> (if revoked: <32-byte-revocation-puzzle-preimage>)</code>
+
+*Since the digest algorithm is double sha256, even for a very large number of revocation, this index would be very cheap to verify.
```

Comment:
?

Reply:
Updated, to state that it is in contrast to, for example, having revocations that are based upon a cryptographic signature.